crates/astria-core/src/protocol/transaction/v1alpha1/mod.rs

```diff
@@ -102,6 +103,11 @@ impl SignedTransaction {
         Sha256::digest(bytes).into()
     }
 
+    #[must_use]
+    pub fn id(&self) -> TransactionId {
+        hex::encode(self.sha256_of_proto_encoding()).into()
```

Comment:
With this we can remove `sha256_of_proto_encoding`. Just rename `s/sha256_of_proto_encoding/id` and let it return a `TransactionId`.

Also note my comment above: don't turn this into a hex-encoding string at this point.